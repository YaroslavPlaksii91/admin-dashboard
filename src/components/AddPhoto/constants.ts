export const IMAGE_SIZE = Object.freeze({
  SMALL: 44,
  BIG: 60,
});
