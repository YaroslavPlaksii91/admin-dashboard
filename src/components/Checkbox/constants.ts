export const CHECKBOX_TEST_ID = 'checkbox';
