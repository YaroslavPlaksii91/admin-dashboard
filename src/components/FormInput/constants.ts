export const TOGGLE_BUTTON_TEST_ID = 'toggleButton';
