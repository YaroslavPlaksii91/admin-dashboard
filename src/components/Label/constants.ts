export const LABEL_TYPES = Object.freeze({
  ROUND: 'round',
  SQUARE: 'square',
});

export const LABEL_COLORS = Object.freeze({
  DEFAULT: 'default',
  SUCCESS: 'success',
  WARNING: 'warning',
  ERROR: 'error',
});
