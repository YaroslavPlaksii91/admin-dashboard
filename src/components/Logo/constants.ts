export const LOGO_TYPES = Object.freeze({
  VERTICAL: 'vertical',
  HORIZONTAL: 'horizontal',
});
