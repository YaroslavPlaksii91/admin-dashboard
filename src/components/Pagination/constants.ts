export const ROWS_PER_PAGE = 8;
export const ROWS_PER_PAGE_OPTIONS = [8, 12, 16, 20];
