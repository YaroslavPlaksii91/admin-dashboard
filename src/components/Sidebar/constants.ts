export const USERS_PAGES = Object.freeze([
  Object.freeze({
    name: 'Overview',
    icon: '/src/assets/icons/sprite.svg#icon-overview',
    link: '/overview',
  }),
  Object.freeze({
    name: 'Tickets',
    icon: '/src/assets/icons/sprite.svg#icon-tickets',
    link: '/tickets',
  }),
  Object.freeze({
    name: 'Ideas',
    icon: '/src/assets/icons/sprite.svg#icon-ideas',
    link: '/ideas',
  }),
  Object.freeze({
    name: 'Contacts',
    icon: '/src/assets/icons/sprite.svg#icon-contacts',
    link: '/contacts',
  }),
  Object.freeze({
    name: 'Agents',
    icon: '/src/assets/icons/sprite.svg#icon-agents',
    link: '/agents',
  }),
  Object.freeze({
    name: 'Articles',
    icon: '/src/assets/icons/sprite.svg#icon-articles',
    link: '/articles',
  }),
]);

export const USERS_SETTINGS = Object.freeze([
  Object.freeze({
    name: 'Settings',
    icon: '/src/assets/icons/sprite.svg#icon-settings',
    link: '/settings',
  }),
  Object.freeze({
    name: 'Subscription',
    icon: '/src/assets/icons/sprite.svg#icon-subscription',
    link: '/subscription',
  }),
]);
