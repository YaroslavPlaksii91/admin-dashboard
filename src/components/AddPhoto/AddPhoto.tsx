import { FC, useState, ChangeEvent } from 'react';
import {
  TextField,
  Box,
  Button,
  SvgIcon,
  FormLabel,
  Typography,
} from '@mui/material';

import { FileInputProps } from './types';
import { IMAGE_SIZE } from './constants';

export const AddPhoto: FC<FileInputProps> = ({
  label,
  name,
  errors,
  register,
  size,
}) => {
  const [imagePreviewUrl, setImagePreviewUrl] = useState<string>('');

  const handleFileChange = (event: ChangeEvent<HTMLInputElement>) => {
    const file = event.target.files?.[0];
    if (file) {
      const imageUrl = URL.createObjectURL(file);
      setImagePreviewUrl(imageUrl);
    }
  };

  const shouldShowError = errors && errors[name] && !imagePreviewUrl;
  const isImageBig = size === IMAGE_SIZE.BIG;

  return (
    <Box sx={{ textAlign: 'start', mb: isImageBig ? '46px' : 0 }}>
      <TextField
        type="file"
        id={name}
        {...register}
        onChange={handleFileChange}
        sx={{
          position: 'absolute',
          width: '1px',
          height: '1px',
          opacity: 0,
        }}
      />
      <FormLabel
        htmlFor={name}
        sx={{
          display: 'flex',
          alignItems: 'center',
          fontSize: '14px',
          fontWeight: 600,
          lineHeight: '20px',
          letterSpacing: 0.2,
          color: 'mainTextColor',
        }}
      >
        <Button
          variant="outlined"
          component="span"
          sx={{
            width: size,
            height: size,
            border: 'none',
            mr: isImageBig ? 6 : 0,
            '&:hover': { border: 'none' },
          }}
        >
          {imagePreviewUrl ? (
            <Box
              sx={{
                width: size,
                height: size,
                borderRadius: '50%',
                overflow: 'hidden',
              }}
            >
              <img
                src={imagePreviewUrl}
                alt="Photo"
                width={size}
                height={size}
              />
            </Box>
          ) : (
            <SvgIcon titleAccess="Add photo" sx={{ width: size, height: size }}>
              <use href="/src/assets/icons/sprite.svg#icon-addPhoto"></use>
            </SvgIcon>
          )}
        </Button>
        {label}
        {shouldShowError && (
          <Typography
            sx={{
              position: 'absolute',
              top: '100%',
              fontSize: 12,
              lineHeight: '20px',
              letterSpacing: 0.3,
              color: '#d32f2f',
            }}
          >
            {errors[name]?.message as string}
          </Typography>
        )}
      </FormLabel>
    </Box>
  );
};
