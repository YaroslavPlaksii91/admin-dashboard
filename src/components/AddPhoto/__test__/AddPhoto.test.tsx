import { cleanup, render, screen } from '@testing-library/react';
import { UseFormRegisterReturn, FieldErrors } from 'react-hook-form';

import { AddPhoto } from '../AddPhoto';
import { IMAGE_SIZE } from '../constants';

describe('Add photo component', () => {
  afterEach(cleanup);

  const label = 'Label';
  const name = 'photo';
  const size = IMAGE_SIZE.BIG;
  const register: UseFormRegisterReturn<any> = {
    onChange: jest.fn(),
    onBlur: jest.fn(),
    name: name,
    ref: null as any,
  };
  const errors: FieldErrors = {};

  it('should be rendered', () => {
    const { asFragment } = render(
      <AddPhoto
        label={label}
        size={size}
        name={name}
        register={register}
        errors={errors}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    render(
      <AddPhoto
        label={label}
        name={name}
        errors={errors}
        register={register}
        size={size}
      />,
    );

    const labelElement = screen.getByText(label);
    const addButton = screen.getByRole('button');

    expect(labelElement).toBeInTheDocument();
    expect(addButton).toBeInTheDocument();
  });

  it('displays the image preview when an image is provided as src prop', () => {
    const imageUrl = 'https://example.com/photo.jpg';

    render(
      <AddPhoto
        label={label}
        name={name}
        errors={errors}
        register={register}
        size={size}
        src={imageUrl}
      />,
    );

    const imagePreview = screen.getByAltText('Photo');

    expect(imagePreview).toBeInTheDocument();
    expect(imagePreview).toHaveAttribute('src', imageUrl);
  });
});
