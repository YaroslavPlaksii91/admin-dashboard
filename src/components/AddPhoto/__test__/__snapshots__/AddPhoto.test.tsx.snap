// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add photo component should be rendered 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-14h66l3"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-g4d8h0-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="photo"
          name="photo"
          type="file"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <label
      class="MuiFormLabel-root MuiFormLabel-colorPrimary css-goj8zy-MuiFormLabel-root"
      for="photo"
    >
      <span
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1nntmli-MuiButtonBase-root-MuiButton-root"
        role="button"
        tabindex="0"
      >
        <svg
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1xzf3w3-MuiSvgIcon-root"
          focusable="false"
          role="img"
          viewBox="0 0 24 24"
        >
          <use
            href="/sprite.svg#icon-addPhoto"
          />
          <title>
            Add photo
          </title>
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      Label
    </label>
  </div>
</DocumentFragment>
`;
